src/sdk/namespace_fs.js

```diff
@@ -346,6 +346,13 @@ class NamespaceFS {
             const file_path = this._get_file_path(params);
             const stat = await nb_native().stat(DEFAULT_FS_CONFIG, file_path);
             console.log(file_path, stat);
+            // directories are not considered objects and therefore we need
+            // to return with error not found.
+            if (stat.isDirectory()) {
```

Comment:
@eshelmarc
```suggestion
            if (isDirectory(stat)) {
```
This should give out the answer if that is a directory or not.
Please make sure to rebase with the master branch, re-run the tests with this addition, and squash the commits prior to merging the PR.